src/telliot_feeds/reporters/tellorflex.py

```diff
@@ -217,46 +210,58 @@ async def get_num_reports_by_id(self, query_id: bytes) -> Tuple[int, ResponseSta
         tip = 0
-
-        if self.datafeed is not None:
-            # Skip fetching datafeed & checking profitability
-            if self.expected_profit == "YOLO":
-                return self.datafeed
-
-            single_tip, status = await self.autopay.get_current_tip(self.datafeed.query.query_id)
-            if not status.ok:
-                logger.warning("Unable to fetch single tip")
-                return None
-
+        datafeed: DataFeed[Any] = self.datafeed  # type: ignore
+        # Skip fetching datafeed & checking profitability
+        if self.expected_profit == "YOLO":
+            return tip
+
+        single_tip, status = await self.autopay.get_current_tip(datafeed.query.query_id)
+        if not status.ok:
+            logger.warning("Unable to fetch single tip")
+        else:
             tip += single_tip
 
-            feed_tip = await get_feed_tip(self.datafeed.query.query_id, self.autopay)
-            if feed_tip is None:
-                logger.warning("Unable to fetch feed tip")
-                return None
-
-            tip += feed_tip
+        feed_tip = await get_feed_tip(datafeed.query.query_id, self.autopay)
+        if feed_tip is None:
+            logger.warning("Unable to fetch feed tip")
         else:
-            suggested_qtag, autopay_tip = await autopay_suggested_report(self.autopay)
+            tip += feed_tip
 
-            if suggested_qtag is None:
-                suggested_qtag = await tellor_suggested_report(self.oracle)
+        return tip
 
+    async def fetch_datafeed(self) -> Optional[DataFeed[Any]]:
+        if self.datafeed:
+            self.autopaytip = await self.rewards()
+            return self.datafeed
+
+        suggested_qtag, autopay_tip = await autopay_suggested_report(self.autopay)
+        if suggested_qtag:
+            self.autopaytip = autopay_tip
+            self.datafeed = CATALOG_FEEDS[suggested_qtag]  # type: ignore
+            return self.datafeed
+
+        if suggested_qtag is None:
+            suggested_qtag = await tellor_suggested_report(self.oracle)
             if suggested_qtag is None:
                 logger.warning("Could not suggest query tag")
                 return None
-
-            if autopay_tip is not None:
-                tip += autopay_tip
-
-            if suggested_qtag not in CATALOG_FEEDS:
+            elif suggested_qtag not in CATALOG_FEEDS:
                 logger.warning(f"Suggested query tag not in catalog: {suggested_qtag}")
                 return None
+            else:
+                self.datafeed = CATALOG_FEEDS[suggested_qtag]  # type: ignore
+                self.autopaytip = await self.rewards()
+                return self.datafeed
+        return None
 
-            self.datafeed = CATALOG_FEEDS[suggested_qtag]  # type: ignore
-
+    async def ensure_profitable(
+        self,
+        datafeed: DataFeed[Any],
+    ) -> ResponseStatus:
+        status = ResponseStatus()
+        tip = self.autopaytip
```

Comment:
If `self.rewards` fails, won't `self.autopaytip` be `None`? Might want to check for that

Reply:
Also, you're calling `self.rewards()` in `fetch_datafeed()`, when I think it makes more sense to call it once when you check for profitability.

Reply:
Roger that